### PR TITLE
Post about AR::Dirty 5.2 changes

### PR DESCRIPTION
**Pull request by mauro-oto**
This PR adds a post about the upcoming changes in Rails 5.2 regarding the `ActiveRecord::Dirty` module. 

Please check it out, thanks!

### CODE REVIEW

File: _posts/2018-02-07-active-record-5-1-api-changes.markdown [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+2.4.2 :020 > user.will_save_change_to_email?
+ => nil
+2.4.2 :021 > user.has_changes_to_save?
+ => false
+```
+
+In conclusion, if you use these methods and are planning to migrate to Rails 5.2
+or even 5.1, you should take these changes into account:
+
+### Must do:
```
Comment (cleicar) [inline _posts/2018-02-07-active-record-5-1-api-changes.markdown]: This block did not appear as a list to me. The text is displayed in a single paragraph.

File: _posts/2018-02-07-active-record-5-1-api-changes.markdown [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+After modifying an object and after saving to the database, or within `after_save`:
+- `attribute_changed?` should now be `saved_change_to_attribute?`
+- `changed?` should now be `saved_changes?`
+- `changes` should now be `saved_changes`
+- `previous_changes` has no replacement, since the behavior for it changes.
+
+### Optional (less ambiguity, more readable, but longer):
```
Comment (cleicar) [inline _posts/2018-02-07-active-record-5-1-api-changes.markdown]: The same goes here

File: _posts/2018-02-07-active-record-5-1-api-changes.markdown [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+[6] pry(#<User>)> saved_change_to_email
+=> ["john@doe.org", "mauro@ombulabs.com"]
+[7] pry(#<User>)> email_before_last_save
+=> "john@doe.org"
+```
+
+The last three methods' behavior remains unchanged between Rails 5.1 and
+Rails 5.2, but the first four emit those warning messages, and their behavior
+changes between these two versions. These changes are best explained in [this commit](https://github.com/rails/rails/commit/16ae3db5a5c6a08383b974ae6c96faac5b4a3c81),
+and the best aspect of it IMHO is explicitness.
```
Comment (etagwerker) [inline _posts/2018-02-07-active-record-5-1-api-changes.markdown]: @mauro-oto I wouldn't use `IMHO` - I would use `in my opinion` directly

File: _posts/2018-02-07-active-record-5-1-api-changes.markdown [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+- `changed?` should now be `saved_changes?`
+- `changes` should now be `saved_changes`
+- `previous_changes` has no replacement, since the behavior for it changes.
+
+### Optional (less ambiguity, more readable, but longer):
+
+After modifying an object and before saving to the database, or within `before_save`:
+- `attribute_changed?` should now be `will_save_change_to_attribute?`
+- `changed?` should now be `has_changes_to_save?`
+- `changes` should now be `changes_to_save`
```
Comment (etagwerker) [inline _posts/2018-02-07-active-record-5-1-api-changes.markdown]: @mauro-oto Maybe you can add a snippet here that mentions our upgrade series of articles and which has a reference to `fastruby.io`

**Comment by mauro-oto** [comment]
@cleicar @etagwerker thanks for the comments! Just addressed them and updated the PR.

**Pull request merged**